equistore/operations/_dispatch.py

```diff
@@ -0,0 +1,121 @@
+    """
+    if isinstance(array, np.ndarray):
+        return np.linalg.norm(array, axis=axis)
+    elif isinstance(array, TorchTensor):
+        return torch.linalg.norm(array, dim=axis)
+    else:
+        raise TypeError(UNKNOWN_ARRAY_TYPE)
+
+
+def dot(array1, array2):
+    """Compute dot product of two arrays.
+    This function has the equivalent bheaviour of  ``np.dot(array1,array2.T)``.
+    For numpy array, it check if the arrays are 2D, in which case it uses the matmul,
+    which is preferred according to the doc.
+    For torch it uses ``torch.dot``
+    if the arrays are 1D, ``torch.matmul`` is used otherwise.
+    """
+    if isinstance(array1, np.ndarray) and isinstance(array2, np.ndarray):
+        shape1 = array1.shape
+        shape2 = array2.shape
+        if len(shape1) == 2 and len(shape2) == 2:
+            return array1 @ array2.T
+        else:
+            return np.dot(array1, array2.T)
+    elif isinstance(array1, TorchTensor) and isinstance(array2, TorchTensor):
+        shape1 = array1.size()
+        shape2 = array2.size()
+        if len(shape1) == 1 and len(shape2) == 1:
+            return torch.dot(array1, array2.T)
+        else:
+            return torch.matmul(array1, torch.Traspose(array2))
+    else:
+        raise TypeError(UNKNOWN_ARRAY_TYPE)
+
+
+def linalg_inv(array, detach=False):
+    """Compute the (multiplicative) inverse of a matrix.
+    This function has the equivalent bheaviour of  ``numpy.linalg.inv(array)``.
+    """
+    if isinstance(array, np.ndarray):
+        return np.linalg.inv(array)
+    elif isinstance(array, torch.Tensor):
+        result = torch.linalg.inv(array)
+        if detach:
+            result = result.detach()
+        return result
+    else:
+        raise TypeError(UNKNOWN_ARRAY_TYPE)
+
+
+def solve(array1, array2):
+    """Computes the solution of a square system of linear equations
+    with a unique solution.
+    This function has the equivalent
+    behaviour of  ``numpy.linalg.solve(array1,array2)``.
+    """
+    if isinstance(array1, np.ndarray) and isinstance(array2, np.ndarray):
+        return np.linalg.solve(array1, array2)
+    elif isinstance(array1, TorchTensor) and isinstance(array2, TorchTensor):
+        result = torch.linalg.solve(array1, array2)
+        return result
+    else:
+        raise TypeError(UNKNOWN_ARRAY_TYPE)
+
+
+def lstsq(array1, array2, rcond=None):
+    """
+    Computes a solution to the least squares problem
+    of a system of linear equations.
+    Computes the vector x that approximately
+    solves the equation ``array1 @ x = array2``.
+    This function has the equivalent
+    behaviour of ``numpy.linalg.lstsq(array1,array2)``.
+    """
+    if isinstance(array1, np.ndarray) and isinstance(array2, np.ndarray):
+        return np.linalg.lstsq(array1, array2, rcond=rcond)[0]
+    elif isinstance(array1, TorchTensor) and isinstance(array2, TorchTensor):
+        result = torch.linalg.lstsq(array1, array2)[0]
```

Comment:
I think torch lstsq should have a rcond parameter as well?

Reply:
yes, it does but this is a thing i wanted to talk about.
The value `None` which I the default in both cases is different between torch and NumPy. In particular, the torch default is the “old“ NumPy default.

[NumPy](https://numpy.org/doc/stable/reference/generated/numpy.linalg.lstsq.html): 

> `rcond` float, optional
> Cut-off ratio for small singular values of a. For the purposes of rank determination, singular values are treated as zero if they are smaller than rcond times the largest singular value of a.
 The previous default of -1 will use the machine precision as rcond parameter, the new default will use the machine precision times max(M, N). To silence the warning and use the new default, use rcond=None, to keep using the old behavior, use rcond=-1.


[Pytorch](https://pytorch.org/docs/stable/generated/torch.linalg.lstsq.html):

>  If rcond= None (default), rcond is set to the machine precision of the dtype of A.

In pytorch they suggest to set manually a value as default to avoid changing behaviour in the future: 

> The default value of rcond may change in a future PyTorch release. It is therefore recommended to use a fixed value to avoid potential breaking changes

Do we want to do that?